hugegraph-core/src/main/java/com/baidu/hugegraph/cluster/HugeServerInfo.java

```diff
@@ -0,0 +1,301 @@
+
+        map.put(Graph.Hidden.unHide(P.ID), this.id);
+        map.put(Graph.Hidden.unHide(P.LABEL), P.SERVER);
+        map.put(Graph.Hidden.unHide(P.ROLE), this.role);
+        map.put(Graph.Hidden.unHide(P.LOAD), this.load);
+        map.put(Graph.Hidden.unHide(P.UPDATE_TIME), this.updateTime);
+
+        return map;
+    }
+
+    public static HugeServerInfo fromVertex(Vertex vertex) {
+        HugeServerInfo server = new HugeServerInfo((Id) vertex.id());
+        for (Iterator<VertexProperty<Object>> iter = vertex.properties();
+             iter.hasNext();) {
+            VertexProperty<Object> prop = iter.next();
+            server.property(prop.key(), prop.value());
+        }
+        return server;
+    }
+
+    public <V> boolean suitableFor(HugeTask<V> task, long now) {
+        if (this.updateTime.getTime() + EXPIRED_INTERVAL < now ||
+            this.load() + task.load() > MAX_LOAD) {
+            return false;
+        }
+        return true;
+    }
+
+    public static Schema schema(HugeGraphParams graph) {
+        return new Schema(graph);
+    }
+
+    public static final class P {
+
+        public static final String SERVER = Graph.Hidden.hide("server");
+
+        public static final String ID = T.id.getAccessor();
+        public static final String LABEL = T.label.getAccessor();
+
+        public static final String NAME = "~server_name";
+        public static final String ROLE = "~server_role";
+        public static final String LOAD = "~server_load";
+        public static final String UPDATE_TIME = "~server_update_time";
+
+        public static String unhide(String key) {
+            final String prefix = Graph.Hidden.hide("server_");
+            if (key.startsWith(prefix)) {
+                return key.substring(prefix.length());
+            }
+            return key;
+        }
+    }
+
+    public static final class Schema {
+
+        public static final String SERVER = P.SERVER;
+
+        protected final HugeGraphParams graph;
+
+        public Schema(HugeGraphParams graph) {
+            this.graph = graph;
+        }
+
+        public void initSchemaIfNeeded() {
+            if (this.existVertexLabel(SERVER)) {
+                return;
+            }
+
+            HugeGraph graph = this.graph.graph();
+            String[] properties = this.initProperties();
+
+            // Create vertex label '~server'
+            VertexLabel label = graph.schema().vertexLabel(SERVER)
+                                     .properties(properties)
+                                     .useCustomizeStringId()
+                                     .nullableKeys(P.ROLE, P.LOAD,
+                                                   P.UPDATE_TIME)
+                                     .enableLabelIndex(true)
+                                     .build();
+            this.graph.schemaTransaction().addVertexLabel(label);
+
+            // Create index
+            this.createIndexLabel(label, P.ROLE);
```

Comment:
seems no need to query by role

Reply:
query by role in `initServerInfo` of `com/baidu/hugegraph/cluster/ServerInfoManager.java`